### ISSUE TITLE
Split selection tests out of dom/ranges/Range-mutations.html

### ISSUE THREAD

**Issue by rniwa** [w3c/web-platform-tests#3444]
dom/ranges/Range-mutations.html is a mega test and tests both Range mutation behaviors as well as behaviors of [`addRange`](http://w3c.github.io/selection-api/#widl-Selection-addRange-void-Range-range) and [`getRangeAt`](http://w3c.github.io/selection-api/#widl-Selection-getRangeAt-Range-unsigned-long-index).

We should split selection related tests into a separate file since the selection API is still in flux.

**Comment by rniwa** [comment]
@ayg @Ms2ger @travisleithead @annevk

**Comment by cdumez** [comment]
+1

**Comment by cdumez** [comment]
I am willing to prepare the pull request if people agree to this.

**Comment by annevk** [comment]
Seems fine to me, would be nice to hear from @ayg, but I suspect he's ok with it too.

**Comment by ayg** [comment]
Sounds good to me as long as it doesn't cause unnecessary code duplication.

**Comment by ayg** [comment]
On second thought, looking at the actual test -- what's the point of this change?  Are we really ever going to want selections to behave differently from ranges here?  Unless I'm missing something, I think that we should split the test only if there's an actual spec change that makes them behave differently.  Otherwise we just create the opportunity for unwanted inconsistency.

Is the problem just that some browsers fail all the selection variants of the tests because they don't implement .addRange() per spec, and it causes a lot of spam/makes them look bad/etc.?  If so, I'd prefer to change the test to check the range returned by .getRangeAt() and compute expected results based on that instead of the range passed in, so Blink/WebKit pass.  There's no actual behavior difference here that warrants splitting up the code.

**Comment by cdumez** [comment]
Yes, my main issue with the test as it is is that WebKit is failing half of the checks due to a non DOM related bug. I would also be happy to update the test as @ayg suggests instead of splitting.
